### PR TITLE
Add example for analyzing national water model dataset

### PR DESCRIPTION
**Pull request by scharlottej13**

### CODE REVIEW

**Comment by mrocklin** [comment]
What's the result of this script?  Does it give us the fun image of water-stuff at the county level?  Or is this something else?  

I'm curious about how I would use this in conversation.

**Comment by scharlottej13** [comment]
> What's the result of this script? Does it give us the fun image of water-stuff at the county level? Or is this something else?
> 
> I'm curious about how I would use this in conversation.

To start, I wanted to be able to refer to it from the blog post. But I can hold off until I add the part where it makes a cool animation if we're OK blocking on that before publishing the post (or I can move this example to a different public repo)

**Comment by mrocklin** [comment]
> But I can hold off until I add the part where it makes a cool animation if we're OK blocking on that before publishing the post

This seems like the kind of thing that may or may not happen (although I hope that it does).  I recommend not blocking on it.  I'm in support of pushing this blog soon (it's been in flight for a while).  

> To start, I wanted to be able to refer to it from the blog post.

That's enough motivation for me.  +1

File: national-water-model/xarray-water-model.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import dask
+import flox.xarray
+import fsspec
+import numpy as np
+import rioxarray
+import xarray as xr
+
+
+# optionally run with coiled run
+# coiled run --region us-east-1 --vm-type m6g.xlarge python nwm-aws.py
```
Comment (jrbourbeau) [inline national-water-model/xarray-water-model.py:15]: Looks like the name of the script has changed

```suggestion
# coiled run --region us-east-1 --vm-type m6g.xlarge python national-water-model/xarray-water-model.py
```

File: national-water-model/xarray-water-model.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    name="nwm-1979-2020",
+    region="us-east-1", # close to data
+    n_workers=10,
+    scheduler_vm_types="r7g.xlarge", # ARM instance
+    worker_vm_types="r7g.2xlarge",
+    compute_purchase_option="spot_with_fallback" # use spot, replace with on-demand
+)
+
+client = cluster.get_client()
+client.restart()
```
Comment (jrbourbeau) [inline national-water-model/xarray-water-model.py:27]: Maybe this is meaningful, but my guess is this is left from previous debugging 

```suggestion
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @scharlottej13

Just a couple of minor (non-blocking) comments

File: national-water-model/xarray-water-model.py (hunk 1 above)
Comment (scharlottej13) [reply national-water-model/xarray-water-model.py:15]: ah indeed, thanks!

File: national-water-model/xarray-water-model.py (hunk 2 above)
Comment (scharlottej13) [reply national-water-model/xarray-water-model.py:27]: yup, thx!

**Comment by jrbourbeau** [comment]
Thanks @scharlottej13!

**Comment by emily-d-anderson** [comment]
> This seems like the kind of thing that may or may not happen (although I hope that it does). I recommend not blocking on it. I'm in support of pushing this blog soon (it's been in flight for a while).

Can we publish the blog without having an AWS marketplace link live? (That's about 2 weeks out per https://github.com/coiled/general/issues/172)

**Pull request merged**